### PR TITLE
Game/GameObject: Fix Spellcast for 'Silithyst'

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**
by @Whoami
updates #14637

### CODE REVIEW

**Comment by jackpoz** [comment]
does this break anything ?

**Comment by Killyana** [comment]
Tested for more than 2 weeks and I get nothing broken,  just this must be added to the pr:
``` sql
UPDATE `gameobject_template` SET `faction`=1673 WHERE `entry` IN (181597);

**Comment by Rushor** [comment]
k i will add it today

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -1757,6 +1757,8 @@ void GameObject::Use(Unit* user)
         return;
     }
 
+    sOutdoorPvPMgr->HandleCustomSpell(user->ToPlayer(), spellId, this);
```
Comment (jackpoz) [inline src/server/game/Entities/GameObject/GameObject.cpp]: please call this only if user->ToPlayer() doesn't return NULL, some handlers might expect this parameter never to be NULL

**Pull request closed without merging**